### PR TITLE
Proxy protocol V2 support

### PR DESCRIPTION
**Pull request by carlhoerberg**

### CODE REVIEW

**Comment by carlhoerberg** [comment]
Closes #253

File: src/avalanchemq/connection_info.cr [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require "socket"
+
+module AvalancheMQ
+  struct ConnectionInfo
+    getter src : Socket::IPAddress
+    getter dst : Socket::IPAddress
+    property ssl : Bool = false
```
Comment (baelter) [inline src/avalanchemq/connection_info.cr:7]:
```suggestion
    property? ssl : Bool = false
```

File: src/avalanchemq/connection_info.cr [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    property ssl_verify : Bool = false
```
Comment (baelter) [inline src/avalanchemq/connection_info.cr:8]:
```suggestion
    property ssl_verify? : Bool = false
```

File: src/avalanchemq/client/amqp_connection.cr [hunk 3]
```diff
@@ -1,14 +1,15 @@
 module AvalancheMQ
   module AMQPConnection
-    def self.start(socket, remote_address, local_address, vhosts, users, log, events)
+    def self.start(socket, conn_props, vhosts, users, log, events)
+      remote_address = conn_props.src
```
Comment (baelter) [inline src/avalanchemq/client/amqp_connection.cr:4]:
```suggestion
    def self.start(socket, connection_info, vhosts, users, log, events)
      remote_address = connection_info.src
```

File: src/avalanchemq/client/amqp_connection.cr [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
       log.progname += " client=#{remote_address}"
       socket.read_timeout = 15
       if confirm_header(socket, log)
         if start_ok = start(socket, log)
           if user = authenticate(socket, users, start_ok, log)
             if tune_ok = tune(socket, log)
               if vhost = open(socket, vhosts, user, log)
-                Client.new(socket, remote_address, local_address, vhost, user, events, tune_ok, start_ok)
+                Client.new(socket, conn_props, vhost, user, events, tune_ok, start_ok)
```
Comment (baelter) [inline src/avalanchemq/client/amqp_connection.cr:12]:
```suggestion
                Client.new(socket, connection_info, vhost, user, events, tune_ok, start_ok)
```

File: src/avalanchemq/client/client.cr [hunk 5]
```diff
@@ -110,9 +112,9 @@ module AvalancheMQ
         peer_port:         @remote_address.port,
         name:              @name,
         pid:               @name,
-        ssl:               tls_terminated?,
-        tls_version:       tls_version,
-        cipher:            cipher,
+        ssl:               @connection_info.ssl,
```
Comment (baelter) [inline src/avalanchemq/client/client.cr:115]:
```suggestion
        ssl:               @connection_info.ssl?,
```

File: src/avalanchemq/server.cr [hunk 6]
```diff
@@ -190,11 +202,11 @@ module AvalancheMQ
       end
     end
 
-    def handle_connection(socket, remote_address, local_address)
-      client = Client.start(socket, remote_address, local_address, @vhosts, @users, @log, @events)
+    def handle_connection(socket, conn_props)
+      client = Client.start(socket, conn_props, @vhosts, @users, @log, @events)
```
Comment (baelter) [inline src/avalanchemq/server.cr:206]:
```suggestion
    def handle_connection(socket, connection_info)
      client = Client.start(socket, connection_info, @vhosts, @users, @log, @events)
```

File: src/avalanchemq/server.cr [hunk 7]
```diff
@@ -190,11 +202,11 @@ module AvalancheMQ
...
       if client.nil?
         socket.close
-        @log.info { "Connection failed for remote_address=#{remote_address}" }
+        @log.info { "Connection failed for remote_address=#{conn_props.src}" }
```
Comment (baelter) [inline src/avalanchemq/server.cr:209]:
```suggestion
        @log.info { "Connection failed for remote_address=#{connection_info.src}" }
```

**Review by baelter — APPROVED** [review]
Great!

**Pull request merged**